api/nodes/serializers.py

```diff
@@ -689,10 +689,13 @@ def create(self, validated_data):
                     'visible': contributor.visible,
                 })
                 if not contributor.user.is_registered:
-                    node.add_unregistered_contributor(
-                        fullname=contributor.user.fullname, email=contributor.user.email, auth=auth,
-                        permissions=parent.get_permissions(contributor.user), existing_user=contributor.user,
-                    )
+                    try:
+                        node.add_unregistered_contributor(
+                            fullname=contributor.user.fullname, email=contributor.user.email, auth=auth,
+                            permissions=parent.get_permissions(contributor.user), existing_user=contributor.user,
+                        )
+                    except ValidationError as e:
+                        raise InvalidModelValueError(detail=str(e))
```

Comment:
From running your test, I saw that the error message is in an array, so return the first message in that array! It's helpful to catch the error message in your test to catch these sort of things.
```json
{
    "errors": [
        {
            "source": {},
            "meta": {},
            "detail": "[u'Unregistered contributor email address domain is blacklisted.']"
        }
    ],
    "meta": {
        "version": "2.0"
    }
}
```